lib/Makefile

```diff
@@ -43,7 +43,7 @@ LIBVER := $(shell echo $(LIBVER_SCRIPT))
 
 DESTDIR?=
 PREFIX ?= /usr/local
-CPPFLAGS= -DXXH_NAMESPACE=LZ4_
+CPPFLAGS= -DXXH_NAMESPACE=LZ4_ -DLZ4_DLL_EXPORT=1
```

Comment:
`dllexport` is required in Visual C++ but not required in MinGW.

I enable it for MinGW to use `dllimport`. If we use `dllimport`, we need to use the current approach or the following approach:

```c
#if defined(_WIN32) && !defined(__MINGW32__)
#  if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
#    define LZ4LIB_API __declspec(dllexport)
#  else
#    define LZ4LIB_API __declspec(dllimport)
#  endif
#else
#  define LZ4LIB_API
#endif
```

`dllimport` isn't required. It's just for optimization: https://blogs.msdn.microsoft.com/russellk/2005/03/20/lnk4217/

> it is purely a performance optimization

If we don't use `dllimport`, we can use more simple definition like Zstd:

```c
#if defined(_WIN32) && defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
#  define LZ4LIB_API __declspec(dllexport)
#else
#  define LZ4LIB_API
#endif
```

If you prefer to remove `dllimport`, i"ll do it.